sycl/include/sycl/handler.hpp

```diff
@@ -1098,6 +1100,8 @@ class __SYCL_EXPORT handler {
   // supported gcc version is bumped.
   std::tuple<std::array<size_t, 3>, bool> getMaxWorkGroups_v2();
 
+  detail::device_impl &getDeviceImpl();
```

Comment:
I wanted to add that when working with `postProcess` refactoring, but ultimately decided it was a wrong abstraction. Wouldn't be surprised if the same is true here.